File: ios/DeviceActivityManager/DeviceActivityReportManager.m
//
//  DeviceActivityReportManager.m
//  whatssub
//
//  Created by Choi on 9/20/24.
//

#import "React/RCTViewManager.h"

@interface

RCT_EXTERN_MODULE(DeviceActivityReportManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(segmentInterval, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(users, NSString)
RCT_EXPORT_VIEW_PROPERTY(devices, NSArray)

@end

File: ios/DeviceActivityManager/DeviceActivityReportManager.swift
//
//  DeviceActivityReportManager.swift
//  whatssub
//
//  Created by Choi on 9/20/24.
//

import SwiftUI
import UIKit
import React

@available(iOS 16.0, *)
@objc(DeviceActivityReportManager)
class DeviceActivityReportManager: RCTViewManager {
  override class func requiresMainQueueSetup() -> Bool {
    return true
  }

  override func view() -> UIView! {
    DeviceActivityHostingView()
  }
}
